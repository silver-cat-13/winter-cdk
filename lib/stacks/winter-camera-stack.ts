import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { WinterModelConstruct } from '../constructs/winter-model-construct';

export class WinterCameraStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new WinterModelConstruct(this, 'WinterModelConstruct', {
      accountId: props?.env?.account!,
    });
  }
}