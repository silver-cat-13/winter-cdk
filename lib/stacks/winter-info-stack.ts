import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';

export class WinterInfoStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // table to manage data for each aÃ±oÃ±i like date of birth, adoption date, ...
    const winterBasicInfoTable = new dynamodb.Table(this, id, {
      billingMode: dynamodb.BillingMode.PROVISIONED,
      tableName: 'WinterBasicInfo',
      readCapacity: 1,
      writeCapacity: 1,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      partitionKey: {name: 'name', type: dynamodb.AttributeType.STRING},
      pointInTimeRecovery: true,
    });

    // ðŸ‘‡ configure auto scaling on table
    const writeAutoScaling = winterBasicInfoTable.autoScaleWriteCapacity({
      minCapacity: 1,
      maxCapacity: 2,
    });
  
    // ðŸ‘‡ scale up when write capacity hits 75%
    writeAutoScaling.scaleOnUtilization({
      targetUtilizationPercent: 75,
    });
  }
}