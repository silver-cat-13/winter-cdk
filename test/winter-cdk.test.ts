import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { WinterCameraStack } from '../lib/stacks/winter-camera-stack';
import { WinterInfoStack } from '../lib/stacks/winter-info-stack';

test('Test WinterCameraStack', () => {
  const app = new cdk.App();
    // WHEN
  const stack = new WinterCameraStack(app, 'WinterCameraStackTest');
    // THEN
  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});

test('Test WinterInfoStack', () => {
    const app = new cdk.App();
      // WHEN
    const stack = new WinterInfoStack(app, 'WinterInfoStackTest');
      // THEN
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });