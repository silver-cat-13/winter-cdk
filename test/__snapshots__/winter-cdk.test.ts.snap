// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test WinterCameraStack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "WinterModelConstructwintermodeldata2E22608D": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": "winter-model-data",
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`Test WinterInfoStack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "WinterInfoStackTestD7EF747C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "name",
            "AttributeType": "S",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "name",
            "KeyType": "HASH",
          },
        ],
        "PointInTimeRecoverySpecification": {
          "PointInTimeRecoveryEnabled": true,
        },
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 1,
          "WriteCapacityUnits": 1,
        },
        "TableName": "WinterBasicInfo",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "WinterInfoStackTestWriteScalingTargetA5EAA51F": {
      "Properties": {
        "MaxCapacity": 2,
        "MinCapacity": 1,
        "ResourceId": {
          "Fn::Join": [
            "",
            [
              "table/",
              {
                "Ref": "WinterInfoStackTestD7EF747C",
              },
            ],
          ],
        },
        "RoleARN": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":iam::",
              {
                "Ref": "AWS::AccountId",
              },
              ":role/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_DynamoDBTable",
            ],
          ],
        },
        "ScalableDimension": "dynamodb:table:WriteCapacityUnits",
        "ServiceNamespace": "dynamodb",
      },
      "Type": "AWS::ApplicationAutoScaling::ScalableTarget",
    },
    "WinterInfoStackTestWriteScalingTargetTrackingB8601027": {
      "Properties": {
        "PolicyName": "WinterInfoStackTestWriteScalingTargetTracking0EC68C1E",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": {
          "Ref": "WinterInfoStackTestWriteScalingTargetA5EAA51F",
        },
        "TargetTrackingScalingPolicyConfiguration": {
          "PredefinedMetricSpecification": {
            "PredefinedMetricType": "DynamoDBWriteCapacityUtilization",
          },
          "TargetValue": 75,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
